projects/web-wallet/wallet-connect/0001-design.md

```diff
@@ -0,0 +1,774 @@
+  dAppConnect[dApp: Connect to Relay Server] --> dAppSubscribed[dApp: Subscribed to Topic]
+  dAppSubscribed -- Send encrypted session_proposal --> walletSubscribed
+  walletSubscribed -- Receive session_proposal --> walletReview[Wallet: Approve or Reject]
+  walletReview -- Send encrypted approve_session --> dAppReceiveApprove[dApp: Receive approve_session]
+  dAppReceiveApprove --> dAppSecureCommunication[dApp: Establish secure communication]
+  walletReview --> walletSecureCommunication[Wallet: Establish secure communication]
+```
+
+
+**3. Message Exchange** The wallet and the dApp can now exchange encrypted messages through the relay server using the `symKey` shared through the URI (either via QRCode or by Deep-linking).
+
+**4. Session Termination** Either the wallet or the dApp can terminate the session by sending a disconnection message. The relay server then closes the WebSocket connection.
+
+#### 3. Relay Server Concepts
+
+| Field        | Description                                                                                           |
+|--------------|-------------------------------------------------------------------------------------------------------|
+| Topic        | A string identifier used by clients to publish and subscribe to messages.                             |
+| Subscription | Requested by a client and remains active until the client requests to unsubscribe.                     |
+| Publication  | Requested by a client; the Relay server is responsible for routing the message to its recipient.       |
+| Message      | A utf8 string sent when publishing, to be routed to its recipient.                                    |
+| Tag          | An integer used as metadata to identify the type of message being sent.                               |
+| TTL          | The maximum amount of time (in seconds) a message should be cached in the relay serverif undelivered.                    |
+| Prompt       | A boolean flag used to identify if a message is relevant to registered webhooks.                      |
+
+<a name="overview__is_it_secure"/>
+
+### Is it secure?
+
+The information sent between `dApps` and wallets is encrypted symmetrically with a key that only these two entities possess
+
+The key is sent directly to the wallet from the `dApp`. Whether users scan through the QR code or choose deep-link with the wallet, there are no intermediaries, so as long as the `symKey` is not leaked, the communication is safe.
+
+This key is generated using the `generateRandomBytes32` method, here is its implementation:
+
+```typescript
+export function generateRandomBytes32(): string {
+  const random = randomBytes(KEY_LENGTH);
+  return toString(random, BASE16);
+}
+```
+
+`randomBytes` is a method from a library called `stablelib`, which can be seen [here](https://github.com/StableLib/stablelib/blob/a89a438fcbf855de6b2e9faa2630f03c3f3b3a54/packages/random/random.ts#L18).
```

Comment:
These random bytes are generated by the dApp, right? Any way we can generate it on the wallet?

Reply:
Make sense. UX would be bad. So, dApps can generate unsafe random bytes and expose themselves to attack vectors, right? Maybe we should add a note about it in this design regarding the overall security of this protocol.

Reply:
Added a remark in d268353e86322d41a879579f29a72efe53f654c7

I will also mention this on the discussion I opened on their repository